src/raft.rs

```diff
@@ -1794,7 +1849,10 @@ impl<T: Storage> Raft<T> {
 
     fn restore_raft(&mut self, snap: &Snapshot) -> Option<bool> {
         let meta = snap.get_metadata();
-        if self.raft_log.match_term(meta.get_index(), meta.get_term()) {
+        // Do not fast-forward commit if we are requesting snapshot.
```

Comment:
Why?

Reply:
Because the snapshot may be dropped if there is no further proposal after requesting a snapshot.